pilot/pkg/model/push_context.go

```diff
@@ -168,6 +168,23 @@ func (ps *PushContext) Add(metric *PushMetric, key string, proxy *Proxy, msg str
 	metricMap[key] = ev
 }
 
+// Delete will remove an case from the metric.
+func (ps *PushContext) Delete(metric *PushMetric, key string) {
```

Comment:
can you rename all these functions to MetricAddEndpoint MetricDeleteEndpoint, etc. ? So that its easier to distinguish between metrics vs actual add/delete?

Reply:
+1 ( but can be separate PR)

Reply:
Also not clear we need to delete - just set to 0.

Reply:
Actually this is not metrics, it is `  ProxyStatus map[string]map[string]ProxyPushStatus` . And 

`PushContext.UpdateMetrics` set metrics by counting `ProxyStatus ` fields.  So I can change the comments.